daceml/testing/profiling/binary_utils.py

```diff
@@ -0,0 +1,81 @@
+import ctypes
+
+# Requires the .so be in LD_LIBRARY_PATH.
+import os
+
+import torch
+from dace.codegen import targets, compiler
+from dace.codegen.codeobject import CodeObject
+
+
+def _compile_and_get_utils():
+    cpp_code = """
+#include <cuda.h>
+#include <cuda_runtime.h>
+#include <cuda_profiler_api.h>
+
+// Forward declaration.
+void gpu_wait(double length, cudaStream_t stream);
+
+extern "C" void do_gpu_wait(double length) {
+  gpu_wait(length, nullptr);
+}
+extern "C" void profiler_start() {
+    cudaProfilerStart(); 
+}
+extern "C" void profiler_stop() {
+    cudaProfilerStop(); 
+}
```

Comment:
It would be PyTorch-specific, and I don't know if you want that, but you could replace these with `torch.cuda.cudart().cudaProfilerStart()`/`cudaProfilerStop()`.